### PR TITLE
Restriction parameter to prevent more than 7 parameters in the next PRs

### PR DESCRIPTION
**Pull request by sultan**
i have a PR waiting to be pushed/suggested but i need these methods to be refactored first so i dont fall into more than 7 parameters compiler errors + the new design i suggest makes the methods more easy to maintain.

i know this does not add any more features but i hope this can be usefull for the quality and ease of maintenance there after.

the basics are sending a Restriction parameter instead of 4 (lower/upper bound and accept booleans), thus gaining 3 more free params to send for other features (next PR to filter RCs is waiting for this one to be accepted)

Thank you for your time

### CODE REVIEW

**Comment by ajarmoniuk** [comment]
You can also suppress the checkstyle error by using 

`SuppressWarnings("checkstyle:maxParameterNumber")`

**Comment by sultan** [comment]
Thanks, good to know there are such options !
i find it quite harsh for me to bypass the checkstyle and have like 4 to 5 booleans in the final methods of the next PRs.

I find it easier to digest to cut my PR into two (this one for the refactoring, and another for the exclude milestone/previews feature)

File: src/main/java/org/codehaus/mojo/versions/AbstractVersionsReport.java [hunk 1]
```diff
@@ -234,18 +234,10 @@ protected abstract void doGenerateReport( Locale locale, Sink sink )
...
     {
-        boolean includeSnapshots = this.allowSnapshots;
-        if ( allowingSnapshots )
-        {
-            includeSnapshots = true;
-        }
-        if ( allowingSnapshots )
-        {
-            includeSnapshots = false;
-        }
```
Comment (slawekjaranowski) [inline src/main/java/org/codehaus/mojo/versions/AbstractVersionsReport.java:248]: Interesting ... looks like never was working

**Comment by slawekjaranowski** [comment]
> 
> would it be ok for the team to merge this PR before the next feature, or do you prefer i merge my work into one PR ?

simpler PR easier to review - it is ok for me for splitting

File: src/main/java/org/codehaus/mojo/versions/AbstractVersionsReport.java (hunk 1 above)
Comment (ajarmoniuk) [reply src/main/java/org/codehaus/mojo/versions/AbstractVersionsReport.java:248]: No, not really. I contemplated removing it too, but I saw it being used.

this.allowSnapshots was a `Boolean` value, could also be null. Local `allowingSnaphots` was a primitive `boolean`, but it was a kind of an override for the value in the Mojo.

So, if `allowingSnapshots` method argument was set, the code overrote the value from the Mojo object with the method argument.

Comment (sultan) [reply src/main/java/org/codehaus/mojo/versions/AbstractVersionsReport.java:248]: There are two methods with Boolean overrides. One of them was with the boolean primitive and seems not used. I did not felt removing this one. Might consider marking deprecated ?

**Pull request merged**